### PR TITLE
Added printing patterns in C++

### PR DESCRIPTION
**Pull request by VinayakTekade**
Signed-off-by: VinayakTekade <39454332+VinayakTekade@users.noreply.github.com>

## Proposed Changes
- Added 25 patterns

### CODE REVIEW

File: C++/patterns/binaryHalfPyramid.cpp [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#include <iostream>
+using namespace std;
+
+int main()
+{
+#ifndef ONLINE_JUDGE
+    freopen("../input.txt", "r", stdin);
+    freopen("../output.txt", "w", stdout);
+#endif
```
Comment (aravindvnair99) [inline C++/patterns/binaryHalfPyramid.cpp:9]:
```suggestion
```

CodeChef judging content need not be added

File: C++/patterns/diamondInscribedRectangle.cpp [hunk 2]
```diff
@@ -0,0 +1,56 @@
+#include <iostream>
+using namespace std;
+
+int main()
+{
+#ifndef ONLINE_JUDGE
+    freopen("../input.txt", "r", stdin);
+    freopen("../output.txt", "w", stdout);
+#endif
```
Comment (aravindvnair99) [inline C++/patterns/diamondInscribedRectangle.cpp:9]:
```suggestion
```

**Review by aravindvnair99 — CHANGES_REQUESTED** [review]
@VinayakTekade Output doesn't print to console.

**Comment by VinayakTekade** [comment]
@aravindvnair99 Removed it. That's just a preference I have to take input and output from txt files.

**Review by aravindvnair99 — CHANGES_REQUESTED** [review]
> @aravindvnair99 Removed it. That's just a preference I have to take input and output from txt files.

Yes @VinayakTekade, but to keep the repository beginner friendly, we could use console input from the user and output there itself instead of file input and output operations.

Could you also add some helper text for all `cin` as a user might not know that input is required?

**Comment by VinayakTekade** [comment]
@aravindvnair99 Added helper text for inputs.

**Review by aravindvnair99 — APPROVED** [review]

**Pull request merged**